man/systemd.exec.xml

```diff
@@ -2062,11 +2062,13 @@ BindReadOnlyPaths=/var/lib/systemd</programlisting>
       <varlistentry>
         <term><varname>ProtectHostname=</varname></term>
 
-        <listitem><para>Takes a boolean argument or <literal>private</literal>. If enabled, sets up a new UTS namespace
-        for the executed processes. If set to a true value, changing hostname or domainname via
-        <function>sethostname()</function> and <function>setdomainname()</function> system calls is prevented. If set to
-        <literal>private</literal>, changing hostname or domainname is allowed but only affects the unit's UTS namespace.
-        Defaults to off.</para>
+        <listitem><para>Takes a boolean argument or <literal>private</literal>. If enabled, sets up a new UTS
+        namespace for the executed processes. If enabled, a hostname can be optionally specified following a
+        colon (e.g. <literal>yes:foo</literal> or <literal>private:host.example.com</literal>), and the
+        hostname is set in the new UTS namespace for the unit. If set to a true value, changing hostname or
+        domainname via <function>sethostname()</function> and <function>setdomainname()</function> system
+        calls is prevented. If set to <literal>private</literal>, changing hostname or domainname is allowed
+        but only affects the unit's UTS namespace. Defaults to off.</para>
```

Comment:
Thanks! This interface would work for us. Just a thought: have you considered the option where PrivateHostname and ProtectHostname are two independent settings? This would allow e.g. setting a hostname in the unit file, while still preventing the guest to update the hostname using ProtectHostname=yes and the seccomp filters. Enabling either setting would create the new UTS namespace, but both settings can be enabled separately.

Making these separate is not something we need, but maybe this is a simpler end-user experience?

Reply:
You can use `ProtectHostname=yes:foo.example.com`, then the guest sees the specified hostname but cannot update the hostname.

Reply:
I see that now, thanks.

One thing I forgot to mention above (apologies) is that for our use case we need the hostname to be "%i" i.e. the service instance. I think that's why I was originally thinking this might be a separate setting. Would it be possible to add support for specifiers in the hostname?

Reply:
OK. Updated.